src/reducers/composers.js

```diff
@@ -97,13 +97,13 @@ export const addDataToMapComposed = (state, action) => {
   // Update mapState store
   mergedState = {
     ...mergedState,
-    mapState: stateMapConfigUpdater(mergedState.mapState, {payload: {mapState: parsedConfig && parsedConfig.mapState}})
+    mapState: stateMapConfigUpdater(parsedConfig.mapState, {payload: {mapState: mergedState.mapState}})
   };
 
   // Update mapStyle store
   mergedState = {
     ...mergedState,
-    mapStyle: styleMapConfigUpdater(mergedState.mapStyle, {payload: {mapStyle: parsedConfig && parsedConfig.mapStyle}})
+    mapStyle: {...parsedConfig.mapStyle,  ...mergedState.mapStyle}
```

Comment:
So `styleMapConfigUpdater` doesn't need to be called here cause it is called above on mergedState and only modifies not in mapStyle, furthermore it didnt do anything before because it is not deconstructed in `receiveMapConfigUpdater`

Reply:
what do you mean by `cause it is called above on mergedState`? I don't see it being called anywhere else.

styleMapConfigUpdater is imported from `map-style-updaters.js`
https://github.com/uber/kepler.gl/blob/master/src/reducers/map-style-updaters.js#L177-L215

Reply:
updated the PR :thumb: